fhir-bulkdata-webapp/src/main/java/com/ibm/fhir/bulkdata/jbatch/export/patient/PatientExportPartitionMapper.java

```diff
@@ -61,13 +74,46 @@ public PartitionPlan mapPartitions() throws Exception {
         ConfigurationAdapter adapter = ConfigurationFactory.getInstance();
         adapter.registerRequestContext(ctx.getTenantId(), ctx.getDatastoreId(), ctx.getIncomingUrl());
 
+        // Check resourceType needs to be processed
+        List<String> target = new ArrayList<>();
+        SystemExportResourceHandler handler = new SystemExportResourceHandler();
+        for (String resourceType : resourceTypes) {
+            FHIRPersistenceHelper fhirPersistenceHelper = new FHIRPersistenceHelper(handler.getSearchHelper());
+            FHIRPersistence fhirPersistence = fhirPersistenceHelper.getFHIRPersistenceImplementation();
+            Map<String, List<String>> queryParameters = new HashMap<>();
+            Class<? extends Resource> rt = ModelSupport.getResourceType(resourceType);
+            FHIRSearchContext searchContext = handler.getSearchHelper().parseQueryParameters(rt, queryParameters);
+            searchContext.setPageSize(1);
+            searchContext.setPageNumber(1);
```

Comment:
would be a little faster if you skip the count (`_total=none`).  that count query can be expensive if there are no search parameters.

Reply:
I switched to the changes API. It's lightly more effective here.